processor/src/lib.rs

```diff
@@ -443,4 +443,43 @@ where
             self.advice_provider,
         )
     }
+
+    // INTERNALS
+    // --------------------------------------------------------------------------------------------
+
+    /// Returns the process system.
+    #[cfg(feature = "internals")]
+    pub fn system(&self) -> &System {
```

Comment:
small nit: all these fns should be `const` so we can use them as static variables